### DISCUSSION TITLE
No updated pose for CF cf1.And rviz shows that the plane has taken off.

### DISCUSSION THREAD

**Discussion by wangmenghetrue** [USC-ACTLab/crazyswarm#510] [Q&A]
Hi! When hover_swarm.launch is started, the rviz page shows that the plane has taken off. But at this time I am not running any scripts. Just run the hover_swarm.launch startup file. The terminal displays No updated pose for CF cf1.
![IMG_0743](https://user-images.githubusercontent.com/49330108/138425521-0dde729e-454f-42bb-b94f-86326fcbc370.JPG)
![IMG_0744](https://user-images.githubusercontent.com/49330108/138425666-cec40e67-7f9b-4f72-8784-7568170e3e8c.JPG)
Then I continue to run the hello_world.py script. The plane fell to the ground instantly after taking off. I have been stuck with this problem for a long time. I really hope to get your help!
![IMG_0745](https://user-images.githubusercontent.com/49330108/138426218-7436b43a-b35e-443a-96dd-d2036df36c18.JPG)
![IMG_0746](https://user-images.githubusercontent.com/49330108/138426292-048f1671-b9a7-46d5-bc9f-32bff7a7a920.JPG)
My hover_swarm.launch is shown below:
![IMG_0749](https://user-images.githubusercontent.com/49330108/138427067-b63621ae-a843-419e-87a7-ca0036a64bd3.JPG)

**Comment by wangmenghetrue** [comment]
I try to run mocao_helper.launch.
![Screenshot from 2021-10-22 21-07-30](https://user-images.githubusercontent.com/49330108/138463837-726ba21d-34b3-44a7-bbda-fa3d27b0fd0d.png)
![Screenshot from 2021-10-22 21-07-55](https://user-images.githubusercontent.com/49330108/138463889-a7b772e1-ad0a-451b-bff8-aca5ed37c06d.png)
I think it may be the frame rate problem of the motion capture system that causes the situation shown in the figure above. Then I increased the frame rate of the motion capture system lens, and the same situation occurred. However, there is a good sign. The plane has been able to take off normally. However, the display in rviz is still wrong. The inability to reflect is really the flight status of the aircraft.
![Screenshot from 2021-10-22 21-13-36](https://user-images.githubusercontent.com/49330108/138464557-09ced594-7c05-4f3a-ae8c-c4a4929ceeb7.png)
![Screenshot from 2021-10-22 20-56-50](https://user-images.githubusercontent.com/49330108/138465713-c6184db8-7076-4f20-b11d-80081d1d026e.png)

**Comment by whoenig** [comment]
* Your object needs to be called "cf1" not "he001"
* Your VRPN server seems to report really high numbers (20m). Perhaps you need to configure it to report in SI units?

**Comment by wangmenghetrue** [comment]
And how to solve this problem in RVIZ? Launch hover_swarm. Launch, rVIz's planes are already in the sky.

**Comment by whoenig** [comment]
This is not a bug/problem, since it is simply showing the position received over VRPN.

**Comment by wangmenghetrue** [comment]
Ok! Thank you very much！I will revise it tomorrow.

**Comment by Zolony** [comment]
I met the same problem above and I don't know how to solve it yet. I have changed my object name to cf1/cf2, but it is still showing "[ WARN] [1638459450.388917491]: No updated pose for motion capture object cf1",  and rostopic list shows there are only topic name cf1/cmd_*，no cf1/pose. 


Showing below:

1. roslaunch crazyswarm hover_swarm.launch

[ INFO] [1638459685.668835290]: [cf1] Update parameters
[ INFO] [1638459685.725385516]: Update params: 0.071414 s
[ INFO] [1638459685.725645017]: Started 1 threads
[ WARN] [1638459685.736787117]: No updated pose for motion capture object cf1
tracker: cf1
tracker: cf2
[ WARN] [1638459685.737427207]: No updated pose for motion capture object cf1
[ WARN] [1638459685.759275898]: Latency too high! Is 0.011913 s.
[ WARN] [1638459685.769758653]: No updated pose for motion capture object cf1
[ WARN] [1638459685.770244969]: No updated pose for motion capture object cf1
[ WARN] [1638459685.780783573]: No updated pose for motion capture object cf1
[ WARN] [1638459685.790928964]: No updated pose for motion capture object cf1
[ WARN] [1638459685.791171838]: No updated pose for motion capture object cf1
[ WARN] [1638459685.801723779]: No updated pose for motion capture object cf1

2.rostopic list
 
/cf1/cmd_full_state
/cf1/cmd_hover
/cf1/cmd_position
/cf1/cmd_stop
/cf1/cmd_vel
/cf1/cmd_velocity_world
/clicked_point
/initialpose
/joy
/move_base_simple/goal
/pointCloud
/rosout
/rosout_agg
/tf
/tf_static
/virtual_interactive_object

**Reply by whoenig** [reply]
What is the output of `mocap_helper.launch` in your case?

**Reply by Zolony** [reply]
ROS_MASTER_URI=http://localhost:11311

process[mocap_helper-1]: started with pid [29911]
frame 0:
    points:
    rigidBodies:
frame 1:
    points:
    rigidBodies:
frame 2:
    points:
    rigidBodies:
frame 3:
    points:
    rigidBodies:
frame 4:
    points:
    rigidBodies:
frame 5:

**Reply by whoenig** [reply]
Which motion capture system are you using? If optitrack, did you enable to stream assets in your Motive settings?

**Reply by hit-JB** [reply]
I use the vicon motion cap and I have encountered the same problem ,But the mocap_help runs will,I Can't see the crazyflie at the rviz and the terminal shows the info blow:
`[ WARN] [1639884992.966357304]: No updated pose for CF cf1 for 1639884992.966298 s. [ WARN] [1639884992.976571012]: No updated pose for CF cf1 for 1639884992.976451 s. [ WARN] [1639884992.986326886]: No updated pose for CF cf1 for 1639884992.986274 s. [ WARN] [1639884992.996349832]: No updated pose for CF cf1 for 1639884992.996289 s. [ WARN] [1639884993.006453691]: No updated pose for CF cf1 for 1639884993.006391 s. [ WARN] [1639884993.016369565]: No updated pose for CF cf1 for 1639884993.016309 s. [ WARN] [1639884993.026424988]: No updated pose for CF cf1 for 1639884993.026348 s. [ WARN] [1639884993.036433558]: No updated pose for CF cf1 for 1639884993.036373 s. [ WARN] [1639884993.046464267]: No updated pose for CF cf1 for 1639884993.046405`
Could you give me some advice

**Reply by kaushikbalasundar** [reply]
I am running into a similar issue. I'm also using VRPN and named my rigid body to 'cf1', and set the streaming ID to 1. However, I am getting the 'No updated pose for CF cf1' warning. I am able to see cf1 in Rviz however.

**Reply by whoenig** [reply]
Please open a separate discussion, unless you have the exact same configuration (motion capture type etc.)

**Comment by wangmenghetrue** [comment]
When you run mocap_help, is there data output on the terminal? Does this warning keep appearing? Are there screenshots or photos of the screen?




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Wolfgang ***@***.***&gt;; 
发送时间: 2021年12月19日(星期天) 下午5:50
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [USC-ACTLab/crazyswarm] No updated pose for CF cf1.And rviz shows that the plane has taken off. (Discussion #510)






我也遇到了类似的问题。我也在用 VR PN ，将我的刚体命名为‘ cf 1 ’，并将流 ID 设置为 1 。但是，我得到了“ CF cf 1 没有更新姿势”的警告。

—
直接回复此邮件，在 GitHub 上查看或取消订阅.
GitHub Mobile 上的分流通知监督办或机器人. 
***@***.***与&gt;

**Reply by kaushikbalasundar** [reply]
Yes there is output in my case, I am also able to see my CF in RViz. Mocap_helper is giving me the position of the CF1 as a rigid body. 

![cf_vrpn_issue_rviz_1](https://user-images.githubusercontent.com/44526371/146673030-2677eb61-05b3-4bd2-822b-befe86083709.png)
